packages/runtime/test/src/parser/parser.ts

```diff
@@ -898,4 +900,61 @@ describe("Parser", function () {
       assert.deepEqual(res, expected);
     });
   });
+
+  describe("Gas cost of Opcodes from TEAL file", () => {
+    useFixture("teal-files");
+
+    let interpreter: Interpreter;
+    beforeEach(function () {
+      interpreter = new Interpreter();
+    });
+
+    it("Should return correct gas cost for 'Crypto opcodes' for tealversion 1", async () => {
+      interpreter.tealVersion = 1; // by default the version is also 1
+
+      let op = new Sha256([], 1, interpreter);
+      assert.equal(interpreter.gas, 7);
+
+      interpreter.gas = 0;
+      op = new Keccak256([], 2, interpreter);
+      assert.equal(interpreter.gas, 26);
+
+      interpreter.gas = 0;
+      op = new Sha512_256([], 3, interpreter);
+      assert.equal(interpreter.gas, 9);
+
+      interpreter.gas = 0;
+      // eslint-disable-next-line
+      op = new Ed25519verify([], 4, interpreter);
+      assert.equal(interpreter.gas, 1900);
+
+      interpreter.gas = 0;
+      await parser(getProgram(cryptoFile), interpreter);
+      assert.deepEqual(interpreter.gas, 1942); // 7 + 26 + 9 + 1900
+    });
+
+    it("Should return correct gas cost for 'Crypto opcodes' for tealversion 2", async () => {
```

Comment:
should we test constant map values ? I don't think that are going to change or their is any logic in them.
(just a thought) for logic parts we need more tests.

Reply:
Probably not. I don't see any downside of not doing it. Should i remove?

Reply:
No, it's neutral to me.

Reply:
Ok, will keep then.

Reply:
Ok, will keep then.